### PR TITLE
Update/js all npm packages

### PR DESCRIPTION
**Pull request by enejb**
Not sure if this is the right way to do this. But I went and updated all the packages via 

npm install -g npm-check-updates
ncu -u

#### Changes proposed in this Pull Request:
* Update all the packages to the latest version.
* 

#### Testing instructions:
Does the build still work? 
`yarn build`

### CODE REVIEW

**Comment by oskosk** [comment]
Thanks for doing this Enej! .
I have my doubts about this being straightforward to do right away. A few packages are sensitive according to platforms.

Just wanna add that we should check that everything builds ok in:

1. [ ] Windows
2. [ ] MacOS Sierra
2. [ ] MacOs High Sierra
2. [ ] GNU/Linux
2. [ ] GNU/Linux under our sandboxes
2. [ ] WSL (Windows).

Mostly because a few packages download binaries. the most important one being node-sass. Both the direct dependency and the dependency coming from gulp-sass

**Comment by oskosk** [comment]
I rebased this one because tests were failing due to problems already fixed in master

**Comment by enejb** [comment]
Fixes #8882 
Can we get some more helping testing this
cc: @gititon (MacOS Sierra )
cc: @johnHackworth on Linux
cc: @Viper007Bond on Windows

**Comment by oskosk** [comment]
As a first measure it would be good to have the tests here pass. The js tests aren't passing

There's a lot happening here and things like the upgrade to React 16 are simply failing because the code is not yet compatible with it. At least until #8714 gets merged.

File: package.json [hunk 1]
```diff
@@ -45,106 +45,107 @@
...
+    "es6-promise": "4.2.4",
+    "eslint": "4.18.0",
+    "eslint-config-wpcalypso": "2.0.0",
     "eslint-loader": "^1.1.0",
-    "eslint-plugin-jsx-a11y": "^4.0.0",
-    "eslint-plugin-lodash": "2.3.4",
-    "eslint-plugin-react": "6.4.1",
-    "eslint-plugin-wpcalypso": "3.0.2",
-    "extract-text-webpack-plugin": "3.0.1",
+    "eslint-plugin-jsx-a11y": "^6.0.3",
```
Comment (oskosk) [inline package.json]: This one is repeated. We should probably leave the one in devDependencies

**Comment by enejb** [comment]
I don't think this is a good approach. Probably better to update each package individually. 
This is the current list of packages

```
 babel-core                     6.8.0  →   6.26.0
 babel-eslint                   6.1.2  →    8.2.2
 babel-loader                   7.1.2  →    7.1.4
 classnames                     2.2.1  →    2.2.5
 clipboard                     ^1.7.1  →   ^2.0.0
 css-loader                    0.23.1  →  0.28.11
 del                           ^2.2.0  →   ^3.0.0
 es6-promise                    3.0.2  →    4.2.4
 eslint                         3.8.1  →   4.19.1
 eslint-config-wpcalypso        0.7.1  →    2.0.0
 eslint-loader                 ^1.1.0  →   ^2.0.0
 eslint-plugin-jsx-a11y        ^4.0.0  →   ^6.0.3
 eslint-plugin-lodash           2.3.4  →    2.7.0
 eslint-plugin-react            6.4.1  →    7.7.0
 eslint-plugin-wpcalypso        3.0.2  →    4.0.1
 extract-text-webpack-plugin    3.0.1  →    3.0.2
 gulp-autoprefixer             ^3.0.2  →   ^5.0.0
 gulp-babel                     7.0.0  →    7.0.1
 gulp-clean-css                ^2.0.2  →   ^3.9.3
 gulp-eslint                    3.0.1  →    4.0.2
 gulp-jshint                    2.0.0  →    2.1.0
 gulp-live-server              0.0.28  →   0.0.31
 gulp-open                     ^1.0.0  →   ^3.0.0
 gulp-phplint                  ^0.0.6  →   ^0.5.1
 gulp-phpunit                 ^0.14.0  →  ^0.25.0
 gulp-qunit                    ^1.3.1  →   ^2.0.1
 gulp-sourcemaps               ~2.0.0  →   ~2.6.4
 gulp-tap                      ^0.1.3  →   ^1.0.1
 gulp-uglify                   ^2.1.2  →   ^3.0.0
 history                            3  →        4
 jsdom                         ^9.2.1  →  ^11.6.2
 jsdom-global                  ^2.1.0  →   ^3.0.2
 jshint                         2.9.1  →    2.9.5
 jshint-stylish                ~2.1.0  →   ~2.2.1
 node-sass                      4.5.3  →    4.7.2
 page                           1.7.1  →    1.8.5
 prop-types                   15.5.10  →   15.6.1
 react                        ^15.6.1  →  ^16.2.0
 react-dom                    ^15.6.1  →  ^16.2.0
 react-hot-loader              ^1.3.0  →   ^4.0.0
 react-redux                   ^4.4.1  →   ^5.0.7
 react-router                      ~3  →       ~4
 react-tap-event-plugin        ^2.0.0  →   ^3.0.2
 sass-loader                    6.0.6  →    6.0.7
 style-loader                    0.13  →     0.20
 url-loader                    ^0.5.7  →   ^1.0.1
 webpack                        3.8.1  →    4.2.0
 whatwg-fetch                  ^1.0.0  →   ^2.0.3
 chai                          ^3.5.0  →   ^4.1.2
 commander                      2.3.0  →   2.15.1
 grunt                         ^0.4.2  →   ^1.0.2
 grunt-wp-i18n                 ~0.4.6  →   ~1.0.2
 mocha                         ^2.4.5  →   ^5.0.4
 mockery                       ^1.4.1  →   ^2.1.0
 nock                          ^7.2.2  →   ^9.2.3
 qunitjs                      ~1.20.0  →   ~2.4.1
 react-test-renderer          ^15.5.0  →  ^16.2.0
 sinon                        ^1.17.3  →   ^4.4.8
 sinon-chai                    ^2.8.0  →   ^3.0.0
 webpack-dev-server             2.9.2  →    3.1.1
 cross-env    ^5.0.1  →  ^5.1.4
 gulp-rtlcss  ^1.0.0  →  ^1.2.0
```
that have updates.

**Pull request closed without merging**